### DISCUSSION TITLE
HCC USB HW abstraction layer and support to Tiny USB

### DISCUSSION THREAD

**Discussion by nmesb** [hathach/tinyusb#2452] [General]
I am not an expert on USB, I am currently using Embedded USB from HCC  and would like to see if I can interoperate with TIny USB. How I can seamlessly integrate TIny USB? Is this possible

**Comment by karlp** [comment]
HCC is just "another" usb stack, you would normally _replace_ HCC with tinyusb.  I'm guessing you want more modern features and platforms and support than the (no longer existing) HCC can give you?

**Reply by nmesb** [reply]
Thanks Karl for you answer,
I totally share your recommendation. Indeed, better to use Tiny USB,  the USB HCC architecture looks complex,
